onnxruntime/core/providers/nnapi/nnapi_builtin/builders/impl/base_op_builder.cc

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/providers/nnapi/nnapi_builtin/builders/impl/base_op_builder.h"
+
+namespace onnxruntime {
+namespace nnapi {
+
+bool HasExternalInitializer(const InitializedTensorSet& initializers, const NodeUnit& node_unit) {
+  const auto is_ext_initializer =
+      [&](const NodeArg& node_arg) {
+        const auto& input_name(node_arg.Name());
+        if (!Contains(initializers, input_name))
+          return false;
+
+        const auto& tensor = *initializers.at(input_name);
+        if (tensor.has_data_location() &&
+            tensor.data_location() == ONNX_NAMESPACE::TensorProto_DataLocation_EXTERNAL) {
```

Comment:
Side note: at some point we need to update this to support using initializers directly from the ort format model by checking for kTensorProtoMemoryAddressTag